tools/sof_perf_analyzer.py

```diff
@@ -190,21 +192,24 @@ def process_kmsg_file():
     with open(args.kmsg, encoding='utf8') as f:
         ppln_id = None
         for line in f:
-            if match_obj := re.search(r'Create widget', line):
+            if match_obj := re.search(r"Create (widget|pipeline)", line):
                 span_end_pos = match_obj.span()[1]
                 line_split = line[span_end_pos + 1:].split()
                 widget_name = line_split[0]
-                # In the linux kernel, IDA is used to allocated pipeline widget instance ID,
-                # this ID later is used for pipeline creation, thus becomes pipeline ID in the
-                # firmware. Note that ppln_id variable will be assigned properly at pipeline widget
-                # creation, because it is always the first one to be created before all other
-                # widgets in the same pipeline and pipelines are created sequentially.
+                # with sof kernel PR: https://github.com/thesofproject/linux/pull/4709
+                # pipeline and module creation are using separate print in kernle log.
+                # So, for pipeline case, pipeline instance ID will be extracted from pipeline
+                # logs. For module creation, module instance ID and module ID will be
+                # extracted accordingly.
+                # ppln_id, widget_id and widget instance ID will be combined together to
+                # column COMP_ID as the unique ID for each module inside pipelines.
```

Comment:
keep my original comments, it explains why instance id of pipeline widget is the widget pipeline id for the widgets of the pineline